prow/pjutil/pjutil_test.go

```diff
@@ -1211,6 +1209,111 @@ func TestSpecFromJobBase(t *testing.T) {
 				return nil
 			},
 		},
+		{
+			name: "Verify DecorateExtraRefs default true",
+			jobBase: config.JobBase{
+				UtilityConfig: config.UtilityConfig{
+					DecorationConfig: &prowapi.DecorationConfig{
+						BloblessFetch: boolPtr(true),
+					},
+					ExtraRefs: []prowapi.Refs{
+						{
+							Org:           "true-org",
+							BloblessFetch: boolPtr(true),
+						},
+						{
+							Org:           "false-org",
+							BloblessFetch: boolPtr(false),
+						},
+						{
+							Org: "default-org",
+							BloblessFetch: nil,
+						},
+					},
+				},
+			},
+			verify: func(pj prowapi.ProwJobSpec) error {
+				for _, r := range pj.ExtraRefs {
+					got := r.BloblessFetch
+					want := boolPtr(r.Org != "false-org")
+					if diff := cmp.Diff(want, got); diff != "" {
+						return fmt.Errorf("ExtraRefs BloblessFetch for %s differs (-want +got)\n%s", r.Org, diff)
+					}
+				}
+				return nil
+			},
+		},
+		{
+			name: "Verify DecorateExtraRefs default false",
+			jobBase: config.JobBase{
+				UtilityConfig: config.UtilityConfig{
+					DecorationConfig: &prowapi.DecorationConfig{
+						BloblessFetch: boolPtr(false),
+					},
+					ExtraRefs: []prowapi.Refs{
+						{
+							Org:           "true-org",
+							BloblessFetch: boolPtr(true),
+						},
+						{
+							Org:           "false-org",
+							BloblessFetch: boolPtr(false),
+						},
+						{
+							Org: "default-org",
+							BloblessFetch: nil,
+						},
+					},
+				},
+			},
+			verify: func(pj prowapi.ProwJobSpec) error {
+				for _, r := range pj.ExtraRefs {
+					got := r.BloblessFetch
+					want := boolPtr(r.Org == "true-org")
+					if diff := cmp.Diff(want, got); diff != "" {
+						return fmt.Errorf("ExtraRefs BloblessFetch for %s differs (-want +got)\n%s", r.Org, diff)
+					}
+				}
+				return nil
+			},
+		},
+		{
+			name: "Verify DecorateExtraRefs no decoration preserves existing",
+			jobBase: config.JobBase{
+				UtilityConfig: config.UtilityConfig{
+					ExtraRefs: []prowapi.Refs{
+						{
+							Org:           "true-org",
+							BloblessFetch: boolPtr(true),
+						},
+						{
+							Org:           "false-org",
+							BloblessFetch: boolPtr(false),
+						},
+						{
+							Org: "default-org",
+							BloblessFetch: nil,
+						},
+					},
+				},
+			},
+			verify: func(pj prowapi.ProwJobSpec) error {
+				for _, r := range pj.ExtraRefs {
+					got := r.BloblessFetch
+					var want *bool
+					switch r.Org {
+					case "true-org":
+						want = boolPtr(true)
+					case "false-org":
+						want = boolPtr(false)
+					}
```

Comment:
Shouldn't we also check the "default-org" case here (expect BloblessFetch to be nil)?

Reply:
nil is the default value of want, and every Refs is always checked.

Reply:
> nil is the default value of want
Ah, because you are relying on the default zero-value of the pointer-to-bool in `var want *bool`.

You're correct that nil is already the value here, but it's odd that we don't actually use the `BloblessFetch: nil` line on 1295. It makes me think if line 1295 should be deleted. But then that would make the test case less explicit, which hurts readability even more.

LGTM code-wise, but style-wise I would add a comment about this somewhere.

Reply:
PTAL. Added the "default-org" case explicitly.